etcdserver/server.go

```diff
@@ -702,7 +702,11 @@ func (s *EtcdServer) applyRequest(r pb.Request) Response {
 		switch {
 		case existsSet:
 			if exists {
-				return f(s.store.Update(r.Path, r.Val, expr))
+				if r.PrevIndex == 0 && r.PrevValue == "" {
```

Comment:
use `r.PrevIndex > 0 || r.PrevValue != ""`?
it is always used in this way in this function.